pkg/provider/provider.go

```diff
@@ -350,22 +403,84 @@ func (p *Provider) getKeyVaultObjectContent(ctx context.Context, kvClient *kv.Ba
 
 	switch kvObject.ObjectType {
 	case VaultObjectTypeSecret:
-		return getSecret(ctx, kvClient, vaultURL, kvObject)
+		return getSecret(ctx, kvClient, vaultURL, kvObject, objectVersion)
 	case VaultObjectTypeKey:
-		return getKey(ctx, kvClient, vaultURL, kvObject)
+		return getKey(ctx, kvClient, vaultURL, kvObject, objectVersion)
 	case VaultObjectTypeCertificate:
-		return getCertificate(ctx, kvClient, vaultURL, kvObject)
+		return getCertificate(ctx, kvClient, vaultURL, kvObject, objectVersion)
 	default:
 		err := errors.Errorf("Invalid vaultObjectTypes. Should be secret, key, or cert")
-		return "", "", wrapObjectTypeError(err, kvObject.ObjectType, kvObject.ObjectName, kvObject.ObjectVersion)
+		return "", "", wrapObjectTypeError(err, kvObject.ObjectType, kvObject.ObjectName, objectVersion)
 	}
 }
 
+func (p *Provider) getKeyVaultSecretVersions(ctx context.Context, kvClient *kv.BaseClient, kvObject KeyVaultObject, vaultURL string) (versions []string, err error) {
```

Comment:
Why is this only for secrets and not certs/keys?

Reply:
I'd been working with certs, which gave me the expectation that the secrets api worked for keys and certs; but I just tested that and it appears that certs are just a special case. I'll update this to work with certs and keys as well

Reply:
I made the update to call the appropriate api based on the secret type